compiler/ir/backend.js/src/org/jetbrains/kotlin/ir/backend/js/lower/AutoboxingTransformer.kt

```diff
@@ -71,11 +70,59 @@ class AutoboxingTransformer(val context: JsIrBackendContext) : AbstractValueUsag
             else -> this.type
         }
 
+        return useExpressionAsType(actualType, type)
+    }
+
+
+    private val IrFunctionAccessExpression.target: IrFunction
+        get() = when (this) {
+            is IrConstructorCall -> this.symbol.owner
+            is IrDelegatingConstructorCall -> this.symbol.owner
+            is IrCall -> this.callTarget
+            else -> TODO(this.render())
+        }
+
+    private val IrCall.callTarget: IrFunction
+        get() = symbol.owner.realOverrideTarget
+
+
+    override fun IrExpression.useAsDispatchReceiver(expression: IrFunctionAccessExpression): IrExpression {
+        return if (expression.symbol.owner.dispatchReceiverParameter?.let { icUtils.shouldValueParameterBeBoxed(it) } == true)
+            this.useAs(irBuiltIns.anyType)
+        else
+            this.useAsArgument(expression.target.dispatchReceiverParameter!!)
+    }
+
+    override fun IrExpression.useAsExtensionReceiver(expression: IrFunctionAccessExpression): IrExpression {
+        return this.useAsArgument(expression.target.extensionReceiverParameter!!)
+    }
+
+    override fun IrExpression.useAsValueArgument(
+        expression: IrFunctionAccessExpression,
+        parameter: IrValueParameter
+    ): IrExpression {
+
+        return this.useAsArgument(expression.target.valueParameters[parameter.index])
+    }
+
+
+    override fun IrExpression.useAsVarargElement(expression: IrVararg): IrExpression {
+        return this.useAs(
+            // Do not box primitive inline classes
+            if (icUtils.isTypeInlined(type) && !icUtils.isTypeInlined(expression.type) && !expression.type.isPrimitiveArray())
+                irBuiltIns.anyNType
+            else
+                expression.varargElementType
+        )
+    }
+}
+
+class AutoboxingTransformer(context: JsCommonBackendContext) : AbstractValueUsageLowering(context) {
+    override fun IrExpression.useExpressionAsType(actualType: IrType, expectedType: IrType): IrExpression {
         // // TODO: Default parameters are passed as nulls and they need not to be unboxed. Fix this
-        if (actualType.makeNotNull().isNothing())
-            return this
 
-        val expectedType = type
+        if (expectedType.isNothing())
```

Comment:
Before we checked actualType? Why it's difference?

Reply:
I don't remember 😞 . Apparently reverting this change doesn't break any test so I'll just do that to get it out of scope of this PR.